numpy/polynomial/chebyshev.py

```diff
@@ -1830,6 +1831,71 @@ def chebcompanion(c):
     return mat
 
 
+def chebinterp(func, deg, x=None, args=(), window=None):
+    """Approximate a function using Chebyshev polynomial interpolation.
+
+    Return the Chebyshev series instance that interpolates a given function.
+
+    Parameters
+    ----------
+    func : function
+        The function to be approximated. It must be a function of a single
+        variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+        that can be passed in the `args` parameter.
+    deg : int or 1-D array_like
+        Degree(s) of the fitting polynomials. If `deg` is a single
+        integer all terms up to and including the `deg`'th term are included in
+        the interpolation.
+    x : array_like, shape (M,), optional
+        Defines window of interpolation. The domain of `x` is used as the
+        approximation window if `window` is not given.
+    args : tuple, optional
+        Extra arguments to be used in the function call.
+    window : [beg, end], optional
+        Window to use for the returned series.
+
+    Returns
+    -------
+    new_series : series
+        A series that represents the polynomial interpolation
+        to the function.
+
+    Notes
+    -----
+    A low-pass filter derived from Chebyshev approximation has an
+    equiripple passband and a monotonically decreasing stopband.
+
+    Examples
+    --------
+    >>> from numpy import tanh, pi
+    >>> import numpy.polynomial.chebyshev as C
+    >>> C.chebinterp(lambda x: tanh(x)+0.5, 8, window=[0, 2*np.pi])
+    array([ 1.30198814,  0.3540514 , -0.25209078,  0.14032589, -0.05706644,
+            0.01226391,  0.00367748, -0.00455335])
+
+    .. versionadded:: 1.14.0
+
+    """
+    domain = chebdomain
+    if window is None:
+        if x is None:
+            window = chebdomain
+        else:
+            window = pu.getdomain(x)
+
+    x = chebpts1(deg)
+    xfunc = pu.mapdomain(x, domain, window)
+
+    c = np.zeros(deg)
+    for j in range(deg):
+        for k in range(deg):
+            c[j] += func(xfunc[k], *args)*np.cos(j*np.arccos(x[k]))
```

Comment:
Ah, fair. And faster, too! Good eye, I didn't think about it from that angle. Was mostly just following Numerical Recipes, 3rd ed. §5.8